ArduPlane/GCS_Mavlink.cpp

```diff
@@ -1150,6 +1150,20 @@ void GCS_MAVLINK_Plane::handleMessage(mavlink_message_t* msg)
         break;
     }
 
+    case MAVLINK_MSG_ID_STATUSTEXT:
+    {
+        // ignore any statustext messages not from our GCS:
+        if (msg->sysid != plane.g.sysid_my_gcs) {
+            break;
+        }
+        mavlink_statustext_t packet;
+        mavlink_msg_statustext_decode(msg, &packet);
+        char text[MAVLINK_MSG_STATUSTEXT_FIELD_TEXT_LEN+1+4] = { 'G','C','S',':'};
+        memcpy(&text[4], packet.text, MAVLINK_MSG_STATUSTEXT_FIELD_TEXT_LEN);
```

Comment:
On Fri, 2 Sep 2016, Andrew Tridgell wrote:

> > +        char text[MAVLINK_MSG_STATUSTEXT_FIELD_TEXT_LEN+1+4] = { 'G','C','S',':'};
> don't we need to nul terminate here in case the statustext is full length?

Yes, and I was thinking about that when I wrote it (I did add the "+1" in 
there :-) )

I based this code on my memory of the way {} works, and my reading of 
this: http://en.cppreference.com/w/cpp/language/aggregate_initialization 
seems to tell me that text will be zero-initialised past the "GCS:" bit.

---------
Character arrays

Arrays of character types (char, signed char, unsigned char, char16_t, 
char32_t, wchar_t) can be initialized from an appropriate string literal, 
optionally enclosed in braces. Successive characters of the string literal 
(which includes the implicit terminating null character) initialize the 
elements of the array. If the size of the array is specified and it is 
larger than the number of characters in the string literal, the remaining 
characters are zero-initialized.